src/App.test.jsx

```diff
@@ -1,14 +1,76 @@
-import { render } from '@testing-library/react';
+import { render, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { useDispatch, useSelector } from 'react-redux';
 
 import App from './App';
 
-test('App', () => {
-  const { getByText } = render((
-    <App />
-  ));
+jest.mock('react-redux');
 
-  expect(getByText(/추가/)).not.toBeNull();
+describe('App', () => {
+  let dispatch;
+  beforeEach(() => {
+    dispatch = jest.fn();
+    useDispatch.mockImplementation(() => dispatch);
+  });
```

Comment:
```suggestion
  const dispatch = jest.fn();
  
  beforeEach(() => {
    jest.clearAllMocks();
    useDispatch.mockImplementation(() => dispatch);
  });
```

위와 같이 매 테스트 시작 전에 mocking한 것들을 clear 시켜주도록해서 해결 할 수 있을 것 같네요~

Reply:
아 clear하는 방법이 있었군요. 
```js
jest.clearAllMocks(); // 이것도 좋지만
dispatch.mockClear(); // 여기서는 이렇게하는 것이 어떤가요?

Reply:
말씀하신대로 지금은 dispatch만 clear 해주는게 좋겠군요! 😄